pkg/apis/pipeline/v1beta1/customrun_validation_test.go

```diff
@@ -0,0 +1,357 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/diff"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+)
+
+func TestRun_Invalid(t *testing.T) {
+	invalidStatusMessage := "test status message"
+	for _, c := range []struct {
+		name string
+		run  *v1beta1.CustomRun
+		want *apis.FieldError
+	}{{
+		name: "missing spec",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+		},
+		want: apis.ErrMissingField("spec"),
+	}, {
+		name: "Empty spec",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{},
+		},
+		want: apis.ErrMissingField("spec"),
+	}, {
+		name: "Both spec and ref missing",
+		run: &v1beta1.CustomRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "temp",
+			},
+			Spec: v1beta1.CustomRunSpec{
+				CustomRef:          nil,
+				CustomSpec:         nil,
+				ServiceAccountName: "test-sa",
+			},
+		},
+		want: apis.ErrMissingOneOf("spec.ref", "spec.spec"),
+	}, {
```

Comment:
same here, spec.CustomRef, spec.CustomSpec

Reply:
and other similar test cases :)

Reply:
done.